pkg/agent/lease_counter.go

```diff
@@ -0,0 +1,81 @@
+package agent
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/apiserver-network-proxy/pkg/agent/metrics"
+	"sigs.k8s.io/apiserver-network-proxy/pkg/util"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	coordinationv1 "k8s.io/client-go/kubernetes/typed/coordination/v1"
+)
+
+// A ServerLeaseCounter counts leases in the k8s apiserver to determine the
+// current proxy server count.
+type ServerLeaseCounter struct {
+	leaseClient   coordinationv1.LeaseInterface
+	selector      labels.Selector
+	fallbackCount int
+}
+
+// NewServerLeaseCounter creates a server counter that counts valid leases that match the label
+// selector and provides the fallback count (initially 0) if this fails.
+func NewServerLeaseCounter(k8sClient kubernetes.Interface, labelSelector labels.Selector) *ServerLeaseCounter {
+	return &ServerLeaseCounter{
+		leaseClient:   k8sClient.CoordinationV1().Leases(""),
```

Comment:
What does an empty namespace do here? If this is just using the agent namespace, can we remove the leases-namespace flag?

Reply:
set it to always be kube-system.